src/core/componentSystems/StyleSystem.js

```diff
@@ -14,13 +14,34 @@ export class StyleSystem extends MRSystem {
      */
     constructor() {
         super(false);
-        this.cssUpdateNeeded = true;
+
+        // want this system to run based on the true/false trigger
+        this.needsSystemUpdate = true;
 
         document.addEventListener('panel-mutated', () => {
-            this.cssUpdateNeeded = true;
+            this.needsSystemUpdate = true;
         });
     }
 
+    /**
+     * @function
+     * @description Getter to checks if we need to run this system's update call. Overridden implementation returns true if there are any items in this
+     * systems registry that need to be run AND the default systemUpdateCheck is true
+     * (see [MRSystem.needsSystemUpdate](https://docs.mrjs.io/javascript-api/#mrsystem.needssystemupdate) for default).
+     * @returns {boolean} true if the system is in a state where this system is needed to update, false otherwise
+     */
+    get needsSystemUpdate() {
+        return this.registry.size > 0 && super.needsSystemUpdate;
```

Comment:
this needs to be always true and we need to stick with checking `entity.needsStyleUpdate` due to some asynchronous loading of text and event driven styles such as hover and click. `needsSystemUpdate` blocks these.

Reply:
perf - yea i wasnt sure about this one specifically, good to know ~

Reply:
<img width="213" alt="Screenshot 2024-01-22 at 10 59 49 AM" src="https://github.com/Volumetrics-io/mrjs/assets/19628643/65c99b02-feed-4034-b442-e0d97471b8ba">

example of the issue it causes, text loads on a delay so it's ancestors need to be told to update once it's ready.

Reply:
wait - nvm - that's what it's doing i thought - one sec

Reply:
i think this part specifically is what's causing the delayed trigger then:

https://github.com/Volumetrics-io/mrjs/pull/236/files/260bcdc7ad84140e87906599d29b53ada2d6da7e#diff-c755f975e32bd8242038b2c9e13c3c9c01b48119dee283b4898964e3b7b8431eR18-R22

because otherwise this.needsSystemUpdate stays undefined and will always run

Reply:
> going to enforce it 100% for this one - but then what is the point of 
> 
> ```
>        // want this system to run based on the true/false trigger
>         this.needsSystemUpdate = true;
> 
>         document.addEventListener('panel-mutated', () => {
>             this.needsSystemUpdate = true;
>         });
>         ```

Ah this probably isn't necessary anymore.

Reply:
gotcha - cleaning up, and letting it revert back to just the super's implementation then ~